master/pkg/tasks/task.go

```diff
@@ -80,6 +78,7 @@ func (t *TaskSpec) Archives() []container.RunArchive {
 // EnvVars returns all the environment variables.
 func (t TaskSpec) EnvVars() map[string]string {
 	e := map[string]string{
+		"WORKING_DIR": t.Environment.WorkDir(),
```

Comment:
this is the point where I think you should be handling the case where `WorkDir() == nil`

Reply:
Handled this case using a separate function.